server/src/main/java/org/jboss/as/server/Services.java

```diff
@@ -73,8 +73,14 @@ private Services() {
 
     public static final ServiceName JBOSS_PRODUCT_CONFIG_SERVICE = JBOSS_AS.append("product-config");
 
+    @Deprecated
```

Comment:
Please add javadoc with @deprecated stating that this will be removed in the next WildFly Core major or minor release.